### PR TITLE
Fb restart warnings

### LINKED ISSUE #361: Extra restart - checks
When reading a restart file containing extra fields, such as an OASIS coupled configuration that sends fields to other models via coupling at t+0 does by default, there is a risk that the fields that are read do not correspond to the right field. This happens when the restart file contains a different set of extra fields than the model is trying to read.

At the moment there is no check for this possibility, and the result is that a coupled run can silently be using the wrong initial fields, which can easily cause a crash. The purpose of this issue is to, when reading a extra fields in a restart file, give a warning if the field is supposed to be read but it is not present, skipping it, and an error if in this same situation the configuration requires this field to be present in the restart because it is a coupling field that is sent to other models at T+0. In principle a coupling field can be initialized to a default value rather than to sensible values obtained by a previous model run, but this often leads to instabilities of the coupled model resulting in a crash. 

This can easily be achieved by writing in the header of the restart file the FLOGRR vector, and comparing this variable with the current value of the variable when reading the restart file.

### PR DESCRIPTION
**Pull request by ukmo-juan-castillo**
# Pull Request Summary
There is a risk that the restart file is not read properly when it contains extra restart fields, if no proper care is taken care. This is avoided in this branch by reading by default all the fields contained in the restart file, and giving a warning if a restart field is requested but it is not present in the restart file.

## Description
When reading a restart file containing extra fields, such as an OASIS coupled configuration that sends fields to other models via coupling at t+0 does by default, there is a risk that the fields that are read do not correspond to the right field. This happens when the restart file contains a different set of extra fields than the model is trying to read.

At the moment there is no check for this possibility, and the result is that a coupled run can silently be using the wrong initial fields, which can easily cause a crash. In this branch we will read by default all the fields that are contained in the restart file, for which we need to write in the restart file the FLOGRR flags indicating what extra restart is actually written.

A small bug fix has been introduced to be able to read more than one extra restart field specified in ww3_shel.inp files. No changes in answers are expected in properly set configurations, apart from warning messages in the output when appropriate.

### Issue(s) addressed

- fixes #361
- fixes noaa-emc/ww3/issues/361

### Check list  
* Is your feature branch up to date with the authoritative repository (NOAA/develop)?
YES
* Make sure you have checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop), [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop) and [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number)
DONE
* Reviewers: @aliabdolali @ukmo-ccbunney 

### Testing
* How were these changes tested?
Regtest ww3_tp2.14.
* Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)
PARTIALLY - the new functionality will provide a warning when using a synthetic restart file and requesting extra restart fields, such as in ww3_tp2.14, OASACM3 test.
* If a new feature was added, was a new regression test added?
YES   - small modification in OASACM4 requesting an extra restart field not present in the restart file used to run this test, causing the output of the appropriate warning.
* Have regression tests been run?
YES - ww3_tp2.14 configuration
* Which compiler / HPC you used to run the regression tests in the PR? 
CrayXC40 at the Met Office, GNU compiler
* Please provide the summary output of matrix.comp (_matrix.Diff.out_, _matrixCompFull.out_ and _matrixCompSummary.out_):    

[matrix.zip](https://github.com/ukmo-waves/WW3/files/6576108/matrix.zip)

New warnings in the ww3_shel.out files if the restart file read does not contain all the requested extra restart fields.

### CODE REVIEW

**Comment by ukmo-juan-castillo** [comment]
This modification will affect all regtests, as it writes a new variable (FLOGRR) to the restart file.

File: model/ftn/ww3_shel.ftn [hunk 1]
```diff
@@ -1258,7 +1258,7 @@
           END IF
           IF (WORDS(7) .EQ. 'T') THEN
             CALL NEXTLN ( COMSTR , NDSI , NDSEN )
-            READ (NDSI,*,END=2001,ERR=2002) FLDRST
+            READ (NDSI,'(A)',END=2001,ERR=2002) FLDRST
```
Comment (ukmo-ccbunney) [inline model/ftn/ww3_shel.ftn:1261]: Why was this changed? Was the list directed read not working correctly for FLDRST?

**Review by ukmo-ccbunney — APPROVED** [review]
All looks very sensible - just one comment from me.

File: model/ftn/ww3_shel.ftn (hunk 1 above)
Comment (ukmo-juan-castillo) [reply model/ftn/ww3_shel.ftn:1261]: The old read line was just reading the first word of the line, so it was only possible to specify one extra restart field. The new read line reads the whole line and store it in FLDRST, which is later processed to find out all the extra restart fields.

**Comment by aliabdolali** [comment]
I ran the regtests. We had the tests with restart files different as expected plus the known non-b4b tests:
```
mww3_test_03
mww3_test_07
ww3_tp2.10
ww3_tp2.14
ww3_tp2.15
ww3_tp2.18
```
@ukmo-ccbunney could you check ww3_tp2.14 and also confirm you are happy with Juan's answer to your question?

[RESTART.zip](https://github.com/NOAA-EMC/WW3/files/6609815/RESTART.zip)

File: model/ftn/ww3_shel.ftn (hunk 1 above)
Comment (ukmo-ccbunney) [reply model/ftn/ww3_shel.ftn:1261]: OK - was this impacting any existing regtests?

Comment (ukmo-juan-castillo) [reply model/ftn/ww3_shel.ftn:1261]: No, this was not impacting any existing regtests. It is now checked in ww3_tp2.14, in particular in the OASACM4 and OASACM5 tests (the first writes the restart file, the second reads it).

Comment (ukmo-ccbunney) [reply model/ftn/ww3_shel.ftn:1261]: Thanks Juan.

**Review by ukmo-ccbunney — APPROVED** [review]
I can confirm that I see the new warnings in the output of the ww3_tp2.14 regression tests.

**Review by aliabdolali — APPROVED** [review]
